app/services/create_order_service.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class CreateOrderService
+  attr_reader :errors, :order, :reward, :address
+
+  def initialize(params:, employee:)
+    @order = Order.new(reward: @reward, reward_snapshot: @reward, address_snapshot: @address, employee: @employee)
+    @reward = Reward.find(params[:reward_id])
+    @employee = employee
+
+    @address = @employee.address || Address.new(employee: @employee)
+    @street = params.dig(:address, :street)
+    @city = params.dig(:address, :city)
+    @postcode = params.dig(:address, :postcode)
+
+    @errors = []
+  end
+
+  def call
+    return false unless items_in_stock?
```

Comment:
If you have to make more checks at the beginning, you could have a little different flow:
```
find_reward && check_items_in_stock && check_funds
return false if @errors.any?

(...)

def check_items_in_stock
    @errors << 'Not enough items in stock' if @reward.number_of_available_items.positive?
    true
end

```

Reply:
Interesting, but the one I have seems to me to be more logical, clear and orderly.